server/src/callbacks.js

```diff
@@ -140,10 +140,43 @@ Empirica.on("batch", async (ctx, { batch }) => {
         .slice(0, 13)}_${config?.batchName}`;
       batch.set("label", batchLabel);
 
-      // set filenames for storing data
-      const scienceDataFilename = `${process.env.DATA_DIR}/batch_${batchLabel}.scienceData.jsonl`;
-      batch.set("scienceDataFilename", scienceDataFilename);
-      fs.closeSync(fs.openSync(scienceDataFilename, "a")); // create an empty datafile
+      // set filenames for storing data
+      const scienceDataFilename = `${process.env.DATA_DIR}/batch_${batchLabel}.scienceData.jsonl`;
+      batch.set("scienceDataFilename", scienceDataFilename);
+      fs.closeSync(fs.openSync(scienceDataFilename, "a")); // create an empty datafile
+      
+      // Validate GitHub repository access before attempting to push data
+      // Skip validation in test mode without required GitHub properties
+      if (process.env.TEST_CONTROLS === "enabled" &&
+        (process.env.GITHUB_PRIVATE_DATA_OWNER === "none" ||
+          process.env.GITHUB_PRIVATE_DATA_REPO === "none" ||
+          process.env.GITHUB_PRIVATE_DATA_BRANCH === "none")) {
+        info("Skipping GitHub repository validation in test mode");
+      } else {
+        const dataRepos = config?.dataRepos || [];
+        if (config?.centralPrereg) {
+          dataRepos.push({
+            owner: process.env.GITHUB_PRIVATE_DATA_OWNER,
+            repo: process.env.GITHUB_PRIVATE_DATA_REPO,
+            branch: process.env.GITHUB_PRIVATE_DATA_BRANCH,
+          });
+        }
+        
+        // Validate data repositories
```

Comment:
Should we still validate the data and prereg repos in test mode? I agree that we should skip the centralPrereg repo, but the others would still make sense, no?